query-builder/src/main/java/com/datastax/oss/driver/internal/querybuilder/insert/DefaultInsert.java

```diff
@@ -62,6 +63,7 @@ public DefaultInsert(
       @Nullable MissingJsonBehavior missingJsonBehavior,
       @NonNull ImmutableMap<CqlIdentifier, Term> assignments,
       @Nullable Object timestamp,
+      @Nullable Object ttl,
```

Comment:
Shouldn't it be specific type instead of Object?

Reply:
In this case, it is possible to set the TTL as a literal, or a BindMarker. They don't share a common Object hierarchy outside of `Object`. Later in the code, where the cql statement is built, the logic checks to see if the TTL value is a bind marker or a literal and builds accordingly.

Reply:
We could introduce yet another wrapper class. But this is private code, local to an internal class. An `instanceof` check is good enough, we don't need to add more clutter.